### PR TITLE
PYTHON-3737 Use __future__ annotations for forward reference type hints

### PR DESCRIPTION
**Pull request by NoahStapp**
The `test/test_comment.py` changes are due to `__future__` annotations explicitly not evaluating all type annotations, causing the test to fail due to comparing the string `"Optional[Any]"` to the evaluated value `Union[Any, None]`. If we want to keep the original, evaluated comparison, we could explicitly evaluate the type annotation as described here: https://docs.python.org/3/library/typing.html#typing.get_type_hints.

### CODE REVIEW

**Review by blink1073 — APPROVED** [review]
LGTM, thanks!  I think the test file changes are fine.

File: pymongo/change_stream.py [hunk 1]
```diff
@@ -96,7 +97,7 @@ class ChangeStream(Generic[_DocumentType]):
     def __init__(
         self,
         target: Union[
-            "MongoClient[_DocumentType]", "Database[_DocumentType]", "Collection[_DocumentType]"
```
Comment (ShaneHarvey) [inline pymongo/change_stream.py:99]: Is this allowed even though the MongoClient import is under `if TYPE_CHECKING:`?

Comment (blink1073) [reply pymongo/change_stream.py:99]: Yes, because it is converted to a string before it is interpreted.

Comment (ShaneHarvey) [reply pymongo/change_stream.py:99]: Neat!

**Review by ShaneHarvey — APPROVED** [review]
LGTM with one ask

File: pymongo/mongo_client.py [hunk 2]
```diff
@@ -2000,10 +2001,10 @@ def get_default_database(
     def get_database(
         self,
         name: Optional[str] = None,
-        codec_options: Optional["bson.CodecOptions[_DocumentTypeArg]"] = None,
+        codec_options: Optional[bson.CodecOptions[_DocumentTypeArg]] = None,
```
[the remark this answers is not in the document]
Comment (NoahStapp) [reply pymongo/mongo_client.py:2004]: The issue being that bson.CodecOptions doesn't render as a type in the docs, right?

[the remark this answers is not in the document]
Comment (ShaneHarvey) [reply pymongo/mongo_client.py:2004]: Right. Not immediately obvious why it doesn't render.

[the remark this answers is not in the document]
Comment (ShaneHarvey) [reply pymongo/mongo_client.py:2004]: Might be related to https://jira.mongodb.org/browse/PYTHON-3604

[the remark this answers is not in the document]
Comment (NoahStapp) [reply pymongo/mongo_client.py:2004]: Done, https://jira.mongodb.org/browse/PYTHON-3741

**Pull request merged**